### PR TITLE
Mulipart body sender

### PR DESCRIPTION
**Pull request by yorekKap**
After proposed changes are approved I will add javadoc and multipart examples to projects with examples.

### CODE REVIEW

File: src/main/java/coresearch/cvurl/io/exception/BadFileException.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package coresearch.cvurl.io.exception;
+
+public class BadFileException extends RuntimeException {
```
Comment (andyislegend) [inline src/main/java/coresearch/cvurl/io/exception/BadFileException.java]: An exception name is poor. It doesn't represent what happened. Smth like MultipartFileFormException suits better

File: src/main/java/coresearch/cvurl/io/multipart/MultipartBody.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    private List<Part> parts;
+
+    private MultipartBody(String boundary, String multipartType, List<Part> parts) {
+        this.boundary = boundary;
+        this.multipartType = multipartType;
+        this.parts = parts;
+    }
+
+    public static MultipartBody create() {
+        return create(UUID.randomUUID().toString());
```
Comment (andyislegend) [inline src/main/java/coresearch/cvurl/io/multipart/MultipartBody.java]: why we need empty file creation?

File: src/main/java/coresearch/cvurl/io/multipart/MultipartBody.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        notNullParam(boundary, "boundary");
+
+        return new MultipartBody(boundary, MultipartType.MIXED, new ArrayList<>());
+    }
+
+    public List<byte[]> asByteArrays() {
+        var result = parts.stream()
+                .flatMap(part -> (Stream<byte[]>) part.asByteArrays(boundary).stream())
+                .collect(Collectors.toList());
+
```
Comment (andyislegend) [inline src/main/java/coresearch/cvurl/io/multipart/MultipartBody.java]: remove those redundant empty spaces over the code plz

File: src/main/java/coresearch/cvurl/io/multipart/MultipartBody.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    public MultipartBody part(Part part) {
+        notNullParam(part, "part");
+
+        this.parts.add(part);
+        return this;
+    }
+
+    public MultipartBody formPart(String name, Part part) {
```
[the remark this answers is not in the document]
Comment (andyislegend) [reply src/main/java/coresearch/cvurl/io/multipart/MultipartBody.java]: why not addPart?

[the remark this answers is not in the document]
Comment (yorekKap) [reply src/main/java/coresearch/cvurl/io/multipart/MultipartBody.java]: 'formPart' here means part that defined by multipart/form-data (https://tools.ietf.org/html/rfc2388)  as opposed to simple 'part' defined by multipart content type (https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html)

File: src/main/java/coresearch/cvurl/io/exception/BadFileException.java (hunk 1 above)
Comment (yorekKap) [reply src/main/java/coresearch/cvurl/io/exception/BadFileException.java]: ok

**Pull request merged**